config/v1/types_infrastructure.go

```diff
@@ -1277,6 +1307,47 @@ type VSpherePlatformTopology struct {
 	Template string `json:"template,omitempty"`
 }
 
+// VSphereFailureDomainAffinity contains the vCenter cluster vm-host group (virtual machine and host types)
+// and the vm-host affinity rule that together creates a affinity configuration for vm-host based zonal.
+// This configuration within vCenter creates the required association between a failure domain, virtual machines
+// and ESXi hosts to create a vm-host based zone.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'HostGroup' ?  has(self.hostGroup) : true",message="when type is HostGroup, hostGroup field must be defined"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type != 'HostGroup' ?  !has(self.hostGroup) : true",message="when hostGroup field is defined, type must be HostGroup"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'HostGroup' ?  has(self.vmGroup) : true",message="when type is HostGroup, vmGroup field must be defined"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type != 'HostGroup' ?  !has(self.vmGroup) : true",message="when vmGroup field is defined, type must be HostGroup"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'HostGroup' ?  has(self.vmHostRule) : true",message="when type is HostGroup, vmHostRule field must be defined"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type != 'HostGroup' ?  !has(self.vmHostRule) : true",message="when vmHostRule field is defined, type must be HostGroup"
+// +union
+type VSphereFailureDomainAffinity struct {
+	// +kubebuilder:validation:Optional
+	// +openshift:validation:FeatureGateAwareEnum:featureGate=VSphereHostVMGroupZonal,enum=HostGroup;ComputeCluster;Datacenter
```

Comment:
Doesn't need to be a feature gate aware enum
As the parent field using this struct is gated, nothing in here needs to worry about being gated